app/controllers/server.rb

```diff
@@ -1,9 +1,108 @@
+require_relative '../models/client'
+require_relative '../models/processor'
+require 'pry'
+
 module RushHour
   class Server < Sinatra::Base
-    not_found do
+    def not_found(error_type)
+      @error_message = error_type
       erb :error
     end
+
+    get "/" do
+      erb :dashboard
+    end
     
-    # Write the rest of your controller code here!
+    get "/sample_sender" do
+      erb :sample_sender
+    end
+
+    get "/sources" do
+      status 200
+      erb :sources
+    end
+
+    get "/sources/" do
+      return (status 400) && not_found(page_not_found)
+      erb :error
```

Comment:
this `not_found(page_not_found)` already returns the same erb you are directly rendering here